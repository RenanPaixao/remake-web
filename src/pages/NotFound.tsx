import { Button, Center, Heading, Image } from '@chakra-ui/react'
import { useNavigate } from 'react-router-dom'
import logo from '../assets/logo.webp'

export const NotFound = () => {
  const navigate = useNavigate()
  return <Center py={20} flexDirection={'column'}>
    <Image src={logo} alt='Remake logo' my={20} />
    <Heading as={'h1'} py={10}>Page Not found! </Heading>
    <Button colorScheme={'blue'} onClick={() => navigate(-1)}>Go back</Button>
  </Center>
}
