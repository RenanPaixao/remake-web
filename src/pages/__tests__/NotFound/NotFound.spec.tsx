import { expect } from 'vitest'
import { customRender, user } from '../../../../tests/test-utils.tsx'
import { NotFound } from '../../NotFound.tsx'

const useNavigateMock = vi.fn()
describe('NotFound', () => {
  beforeEach(() => {
    vi.mock('react-router-dom', async importOriginal => {
      const actual: object = await importOriginal()
      return {
        ...actual,
        useNavigate: () => useNavigateMock
      }
    })
  })

  it('should render', async () => {
    const { container } = customRender(<NotFound/>)

    expect(container).toMatchSnapshot()
  })

  it('should go back when click on go back button', async () => {
    const { getByRole } = customRender(<NotFound/>)

    const goBackButton = getByRole('button', { name: /Go back/i })

    await user.click(goBackButton)

    expect(useNavigateMock).toHaveBeenCalledWith(-1)
  })
})
