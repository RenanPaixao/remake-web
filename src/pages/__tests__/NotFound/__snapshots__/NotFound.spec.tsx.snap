// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > should render 1`] = `
<div>
  <div
    class="css-1h2nyg0"
  >
    <img
      alt="Remake logo"
      class="chakra-image css-ibmrth"
      src="/src/assets/logo.webp"
    />
    <h1
      class="chakra-heading css-gqdkvc"
    >
      Page Not found! 
    </h1>
    <button
      class="chakra-button css-jut409"
      type="button"
    >
      Go back
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
